include/ibamr/IrregularWaveGenerator.h

```diff
@@ -0,0 +1,85 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (c) 2019 - 2019 by the IBAMR developers
+// All rights reserved.
+//
+// This file is part of IBAMR.
+//
+// IBAMR is free software and is distributed under the 3-clause BSD
+// license. The full text of the license can be found in the file
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+/////////////////////// INCLUDE GUARD ////////////////////////////////////
+
+#ifndef included_IrregularWaveGenerator
+#define included_IrregularWaveGenerator
+
+///////////////////////////// INCLUDES ///////////////////////////////////
+
+#include "ibamr/StokesWaveGeneratorStrategy.h"
+
+#include <fstream>
+
+namespace IBAMR
+{
+/*!
+ * \brief Class for generating Irregular waves.
+ */
+class IrregularWaveGenerator : public StokesWaveGeneratorStrategy
+{
+public:
+    IrregularWaveGenerator(const std::string& object_name, SAMRAI::tbox::Pointer<SAMRAI::tbox::Database> input_db);
+
+    /*!
+     * Get surface elevation at a specified horizontal position and time.
+     */
+    double getSurfaceElevation(double x, double time) const;
+
+    /*!
+     * Get velocity component at a specified position and time.
+     */
+    double getVelocity(double x, double z_plus_d, double time, int comp_idx) const;
+
+    /*!
+     * Writing the wave data.
+     */
+    void getWaveData(std::ofstream& ostream) const;
+
+private:
+    /*!
+     * Get wave parameters from input db.
+     */
+    void getFromInput(SAMRAI::tbox::Pointer<SAMRAI::tbox::Database> db);
+
+    /*!
+     * \brief Wave parameters.
+     *
+     * \param d_num_waves  Number of component waves with random phases to be generated (default = 50)
+     * \param d_depth      Depth of water, from sea bed to still water level [m]
+     * \param d_Hs         Significant wave height [m]
+     * \param d_Ts         Significant wave period [s]
+     */
+    int d_num_waves = 50; // default value is set to 50
+    double d_Ts = std::numeric_limits<double>::quiet_NaN(), d_Hs = std::numeric_limits<double>::quiet_NaN();
+
+    /*!
+     * \brief Wave Spectrum parameters.
+     *
+     * \param d_omega_begin     Lowest angular frequency in the spectrum [rad/s]
+     * \param d_omega_end       Highest angular frequency in the spectrum [rad/s]
+     * \param d_wave_spectrum   JONSWAP/Bretschneider wave spectrum.
+     * \param d_omega           Angular frequencies of component waves [rad/s]
+     * \param d_wave_number     Wave number of component waves [2\f$ \pi \f$/m]
+     * \param d_amplitude       Amplitude of component waves [m]
+     * \param d_phase           Phase (random) of component waves [rad]
+     */
```

Comment:
This comment block has the same issue: this will not show up in doxygen correctly. Would you please split this so that there is a single comment block for each class member that needs to be documented?

Reply:
Kaustubh can you send the SAMRAI doxygen that you followed.

Reply:
I am not 100% sure that you cannot make this work for member variables, but I think that Doxygen does want separate annotations for each member variable. I think that you can use `///` so that these can be made one-line annotations, e.g.,
```
    /// Amplitude of component waves [m]
    std::vector<double> d_amplitude;
```
Also, does the implementation require the use of a specific system of units? If not, I would suggest omitting specific units wherever possible. So, e.g.,
```
    /// Amplitude of component waves [length]
    std::vector<double> d_amplitude;
```

Reply:
> I am not 100% sure that you cannot make this work for member variables, but I think that Doxygen does want separate annotations for each member variable. I think that you can use 
 
Ok, let's do one-line annotation then. But it would be nice to confirm this somehow for future usage.  
`///` so that these can be made one-line annotations, e.g.,
> 
> ```
>     /// Amplitude of component waves [m]
>     std::vector<double> d_amplitude;
> ```
> 
> Also, does the implementation require the use of a specific system of units? If not, I would suggest omitting specific units wherever possible. So, e.g.,
> 
> ```
>     /// Amplitude of component waves [length]
>     std::vector<double> d_amplitude;
> ```

Yeah that sounds right. It should be dimensions [length, time] rather than units [m,s].